admin-perspective/olm/install-forcenamespace/install-forcenamespace.md

```diff
@@ -0,0 +1,85 @@
+# Operator-defined Installation Namespace
+
+Operator bundles can define what namespace they should be installed to, so that namespace is recommended as the default to the user. The user can still change the installation namespace if desired.
+
+This design is now surfacing both the watch (availability) namespace and the install namespaces to the user. The installed namespace is initially read-only and defaults to **openshift-operators** when the availability namespace is **all namespaces**, and defaults to the **value from watch namespace dropdown** when the availability namespace is **single** & has a namespace selected. The user has the option to expand the advanced settings to modify installation namespace if desired (and the operator allows.)
+
+This design is not intended to convey every combination of controls that is possible via operator InstallModes, that information is available via [this spreadsheet](https://docs.google.com/spreadsheets/d/19J7IEjORg4-U6nQEwQkxPwe3SJazgpobN2HFKE2rFcc/edit?folder=1y5NhdQNm8b-NeX1I0uN6PHQzhp_NJPh0#gid=0). Controls in this design will be enabled/disabled/hidden as described in the spreadsheet and as conveyed in the screens below and the existing UI’s paradigms. This design is intended to show the location of controls and the interactions between them.
+
+This design also captures the ability to enable cluster monitoring on namespaces that are installed to and the operator requests it. This setting applies to only certain whitelisted namespaces, and the user should be warned of the implications of doing so. Operators can request monitoring be enabled on their install namespace but we will also present this option 
+
+## Installing with no operator-defined namespace (existing behavior)
+
+![Operator namespace not defined all](img/1-1-noDefinedNS-all.png)
+- Installing an operator without a operator-defined namespace functions the same as the existing UI, though now the install namespace is conveyed.
+- Operators available to the entire cluster are installed on **openshift-operators** and is not editable.
+
+![Operator namespace not defined single](img/1-2-noDefinedNS-single.png)
+-A specific namespace may be chosen for the operator to be available to (and installed to.)
+
+![Operator namespace not defined single](img/1-3-noDefinedNS-single.png)
+-The install namespace will match the specific namespace the operator is available to, until the install namespace is manually changed.
+
+![Operator namespace not defined single](img/1-4-noDefinedNS-single.png)
+- The user can **Show Advanced Installed Namespace Options** to modify the namespace the operator is installed to.
```

Comment:
Not really sure if it makes sense to have a **Show Advanced Installed Namespace Options** link, if it will only contain a single dropdown, would incline to show the namespace dropdown directly, without the link.
@spadgett thoughts ?

Reply:
The thought was that most users would not want to change this install namespace and we wanted to hide that complexity from them unless they ask for it. @dmesser

Reply:
Not sure this link is not introducing the extra complexity.

If anything I would suggest to change the name from **Show Advanced Installed Namespace Options** to **Change Installed Namespace**, since the link function is purely to change the namespace, nothing else.

Reply:
@dmesser Do you think there is still worth in showing a read-only version of the install namespace initially, with the user having to take a deliberate **Edit** action, vs just showing the edit controls (dropdowns/radios) all the time?

My feeling now is that there still is, though perhaps the Show/Hide expandable control might not be the best suited for this, and maybe something more like the "one way" **Edit** link I initially proposed might have been a better fit.

Reply:
I think it feels weird to be able to edit something that is read-only.

Reply:
I've updated the design to not use the expandable control, since that was probably heavy-handed for this need. Instead a more subtle **Edit** link would be used to display the controls to change the install namespace.

This means that the typical install case would be something like what is seen in screens 1-1 & 1-3, which is rather similar to the existing UI, just with an additional label conveying the install namespace. These would be the same namespaces that had been used all along, just now more visible to the user. This would be less of a change than showing the dropdown to modify the install namespace all the time, as that would give the user more of an impression they should deliberately modify it.